### ISSUE TITLE
MsgPack::pack can't pack type: 51 while packing Qt-parsed JSON with null values

### ISSUE THREAD

**Issue by wincak** [romixlab/qmsgpack#38]
Qt function QJsonObject::toVariantMap() parses JSON 'null' values as QVariants with QMetaType::Nullptr type. When processing these maps with MsgPack::pack() the library prints error:

> MsgPack::pack can't pack type: 51

(51 is QMetaType::Nullptr)
It seems that qmsgpack expects not Nullptr type QVariants but null QVariants.
pack_p.cpp:
```
quint8 *MsgPackPrivate::pack(const QVariant &v, quint8 *p, bool wr, QVector<QByteArray> &user_data)
{
    QMetaType::Type t = (QMetaType::Type)v.type();
    if (v.isNull() && !v.isValid())
        p = pack_nil(p, wr);
...
```

This seems to be more of an incompatibility than simply a bug but nevertheless causes problems when converting JSON to MsgPack.

Issue can be reproduced with following code:
```
#include <QJsonDocument>
#include <QJsonObject>

#include "msgpack.h"

int main(int argc, char *argv[])
{
    QString json("{ \"test\" : null }");
    QJsonDocument jsonDoc(QJsonDocument::fromJson(json.toUtf8()));
    QVariantMap map = jsonDoc.object().toVariantMap();

    MsgPack::pack(map);
}
```
It's probably possible to write a user packer, but that feels like a workaround.

Tested with Qt 5.12.2 and qmsgpack e3819fc5b3e3d52bbb7b5aaa47319dc09743c30f

**Comment by DexterMagnific** [comment]
Dear developer,

Please handle this issue. Translating JSON to binary format allows some sort of program saved data obfuscation.

**Comment by romixlab** [comment]
Hi! Sorry for late answer, I will look into it soon. Does seem like an easy fix

**Comment by romixlab** [comment]
Fixed a9e429a30fcc906b4bd0a471f22e0b96b7b1528c

**Comment by romixlab** [comment]
Actually as it turned out after CI build which uses pretty old Qt, QMetaType::Nullptr was added after 5.8.0, so I've also added ifdef for that case.
